src/oc/web/components/comments.cljs

```diff
@@ -19,39 +19,112 @@
             [goog.events :as events]
             [goog.events.EventType :as EventType]))
 
-(defn delete-clicked [s]
-  (let [comment-data (first (:rum/args s))]
-    (let [alert-data {:icon "/img/ML/trash.svg"
-                      :action "delete-comment"
-                      :message (str "Delete this comment?")
-                      :link-button-title "No"
-                      :link-button-cb #(do
-                                         (dis/dispatch! [:alert-modal-hide])
-                                         (reset! (::editing? s) false))
-                      :solid-button-title "Yes"
-                      :solid-button-cb #(let [activity-uuid (router/current-activity-id)]
-                                         (dis/dispatch! [:comment-delete activity-uuid comment-data])
-                                         (dis/dispatch! [:alert-modal-hide]))
-                      }]
-      (dis/dispatch! [:alert-modal-show alert-data]))))
+(defn setup-medium-editor [comment-node]
+  (let [config {:toolbar false
+                :anchorPreview false
+                :extensions #js []
+                :autoLink true
+                :anchor false
+                :paste #js {:forcePlainText true}
+                :placeholder #js {:text "Share your thoughts..."
+                                  :hideOnClick true}
+               :keyboardCommands #js {:commands #js [
+                                  #js {
+                                    :command false
+                                    :key "B"
+                                    :meta true
+                                    :shift false
+                                    :alt false
+                                  }
+                                  #js {
+                                    :command false
+                                    :key "I"
+                                    :meta true
+                                    :shift false
+                                    :alt false
+                                  }
+                                  #js {
+                                    :command false
+                                    :key "U"
+                                    :meta true
+                                    :shift false
+                                    :alt false
+                                  }]}}]
+    (new js/MediumEditor comment-node (clj->js config))))
+
+(defn stop-editing [s]
+  (let [medium-editor @(::medium-editor s)]
+    (.destroy medium-editor)
+    (when @(::esc-key-listener s)
+      (events/unlistenByKey @(::esc-key-listener s))
+      (reset! (::esc-key-listener s) nil))
+    (reset! (::medium-editor s) nil)
+    (reset! (::editing? s) false)
+    (dis/dispatch! [:input [:comment-edit] nil])))
+
+(defn cancel-edit
+  [e s c]
+  (.stopPropagation e)
+  (let [comment-field (rum/ref-node s "comment-body")]
+    (set! (.-innerHTML comment-field) (utils/emojify @(::initial-body s) true))
```

Comment:
I am seeing a bug and I don't think you want `::initial-body` here. To recreate the issue edit a comment and save with enter.  After that open the comment and hit escape. You will see the original data.

What you want here is to use the current comment data. This is where you will want to read data from the "Store" to get the current state. Right now that means read from the app state.

Reply:
@thepug addressed the issue, good catch!